MTA-7133-LLM-Proxy-Disconnection-Network-Isolation-Procedures (#381)
migtools/mta-documentation: docs/topics/developer-lightspeed/con_llm-proxy-shutdown-overview.adoc

// Module included in the following assemblies:
//
// * docs/developer-lightspeed-guide/master.adoc

:_mod-docs-content-type: CONCEPT

[id="llm-proxy-shutdown-overview_{context}"]
= LLM proxy shutdown overview

[role="_abstract"]

> The method that you use to shut down the large language model (LLM) proxy depends on how {ProductShortName} is deployed.

Reviewer: ```suggestion
To stop all outbound large language model (LLM) traffic during a security incident, you must shut down the LLM proxy. The method you use to disable the proxy depends on how {ProductShortName} is deployed.
```

[WARNING]
====

> Perform these steps only under the direction of your security or incident-response team. Disabling the proxy immediately severs all in-progress LLM sessions for every connected developer without warning.

Reviewer: ```suggestion
You must perform the proxy shutdown procedures only under the direction of your security or incident-response team. Disabling the proxy immediately severs all in-progress LLM sessions for every connected developer without warning.
```

====

Centralized configuration mode:: In Operator-managed or {ocp-short} deployments, the {ProductShortName} Operator manages the LLM proxy lifecycle. You disable the proxy by setting `kai_llm_proxy_enabled: false` in the Tackle custom resource (CR). The Operator then terminates the proxy pod and does not re-create it.
+
[IMPORTANT]
====
Do not directly scale the proxy `Deployment` object to zero replicas. The {ProductShortName} Operator reconciles managed resources and re-creates any manually scaled-down pods. To permanently disable the proxy, edit the Tackle CR.
====

> Standalone mode:: In standalone mode, there is no central proxy component to disable. Each developer IDE connects directly to the configured public LLM endpoint. Network administrators must block this traffic at the corporate firewall or network policy layer.

Author: ```suggestion
[id="llm-proxy-shutdown-overview_{context}"]
= LLM proxy shutdown overview

[role="_abstract"]
The method that you use to shut down the large language model (LLM) proxy depends on how {ProductShortName} is deployed.

[WARNING]
====
Perform these steps only under the direction of your security or incident-response team. Disabling the proxy immediately severs all in-progress LLM sessions for every connected developer without warning.
====

Centralized configuration mode:: In Operator-managed or {ocp-short} deployments, the {ProductShortName} Operator manages the LLM proxy lifecycle. To disable the proxy, set `kai_llm_proxy_enabled: false` in the Tackle custom resource (CR). The Operator then terminates the proxy pod and does not re-create it.
+
[IMPORTANT]
====
Do not directly scale the proxy `Deployment` object to zero replicas. The {ProductShortName} Operator reconciles managed resources and re-creates any manually scaled-down pods. To disable the proxy permanently, edit the Tackle CR.
====

Standalone mode:: Standalone deployments do not have a central proxy component to disable. Each developer integrated development environment (IDE) connects directly to the configured public LLM endpoint. You must block this traffic at the corporate firewall or network policy layer.
```
